docs/collection.js

```diff
@@ -0,0 +1,279 @@
+     * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of `for-of`}
+     * loops,
+     * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator `...`}
+     * spread operators, and more.
+     * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/iterator Symbol.iterator}
+     *   and the {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#iterable iterable protocol}
+     * @returns {Realm.Collection~Iterator} of each Realm object in the collection
+     * @since 0.11.0
+     * @example
+     * for (let object of collection) {
+     *   // do something with each object
+     * }
+     */
+    [Symbol.iterator]() {}
+
+    /**
+     * Joins all objects in the collection into a string.
+     * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join Array.prototype.join}
+     * @param {string} [separator=","] - A string to separate the return values of the
+     *   `toString()` method being called on each object in the collection.
+     * @returns {string}
+     * @since 0.11.0
+     */
+    join(separator) {}
+
+    /**
+     * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice Array.prototype.slice}
+     * @param {number} [start=0] - The start index. If negative, then the start index will be
+     *   counted from the end of the collection.
+     * @param {number} [end] - The end index. The objects up to, but not including, the end
+     *   index will be include in the return value. If negative, then the end index will be
+     *   counted from the end of the collection. If omitted, then all objects from the start
+     *   index will be included in the return value.
+     * @returns {Realm.Object[]} containing the objects from the start index up to, but not
+     *   including, the end index.
+     * @since 0.11.0
+     */
+    slice(start, end) {}
+
+    /**
+     * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find Array.prototype.find}
+     * @param {function} callback - Function to execute on each object in the collection.
+     *   If this function returns `true`, then that object will be returned by this method.
+     *   This function takes three arguments:
+     *   - `object` – The current object being processed in the collection.
+     *   - `index` – The index of the object being processed in the collection.
+     *   - `collection` – The collection itself.
+     * @param {object} [thisArg] - The value of `this` when `callback` is called.
+     * @returns {Realm.Object|undefined} if the `callback` did not return `true` for any object
+     *   in the collection.
+     * @since 0.11.0
+     */
+    find(callback, thisArg) {}
```

Comment:
It's there for parity with `Array`. I don't think it's dangerous because users will just use `Array.prototype.find` or `for` loops instead if they need to. In some cases, you might need to use this over a query because queries can't do everything, and this method returns early after the first one is found.